src/main/java/com/openshift/internal/restclient/model/Container.java

```diff
@@ -142,13 +143,19 @@ public String getImagePullPolicy() {
 	}
 
 	@Override
-	public void setLifecycle(String lifecycle) {
-		set(node, propertyKeys, LIFECYCLE, lifecycle);
+	public void setLifecycle(ILifecycle lifecycle) {
+		ModelNode lifecycleNode = ModelNode.fromJSONString(lifecycle.toJson());
+		get(node, propertyKeys, LIFECYCLE).set(lifecycleNode);
 	}
 
 	@Override
-	public String getLifecycle() {
-		return asString(node, propertyKeys, LIFECYCLE);
+	public ILifecycle getLifecycle() {
+		ModelNode lifecycle = get(node, propertyKeys, LIFECYCLE);
```

Comment:
I know elsewhere we use this pattern, but recently introduced 'has' for checking if its there, which is what we should use.  The benefit is that it wont add the node to the underlying json if its 'not defined'